### DISCUSSION TITLE
ValueError in dataset transform function

### DISCUSSION THREAD

**Discussion by roybenhayun** [torchgeo/torchgeo#1631] [Q&A]
just a minor question..

usually we give the dataset a transform function i.e., `ds = Sentinel2(eo_root, transforms=sentinel2_preprocess_image)`.
the code in the transform function was based on examples and has `del sample["bbox"]`.
I noticed that when this deletion is not included, getting `ValueError: A frozen dataclass was passed to `apply_to_collection` but this is not allowed.`

the documentation for transform in dataset says only 
> a function/transform that takes an input sample and returns a transformed version

could you please clarify why this ValueError is happening and if anything else is required in the transform function?

**Comment by adamjstewart** [comment]
Can you provide code to reproduce this issue?
